ivy/functional/backends/torch/layers.py

```diff
@@ -372,6 +372,7 @@ def conv3d_transpose(
     return res
 
 
+@auto_shift_device
```

Comment:
I think the decorator should be applied to the `ivy` function rather than the backend impementations, as the behaviour should be consistent across all backends

Reply:
Hi @VedPatwardhan @sherry30, I am using the following example to test the functionality(hope this is the expected scenario):
```python
import ivy

ivy.set_backend("torch")
ivy.set_soft_device_placement(True)

x = ivy.random_uniform(shape=(1, 1, 1, 2), device="gpu:0")
f = ivy.random_uniform(shape=(2, 1, 2, 2), device="cpu:0")

c = ivy.linear(x, f)
```
On calling `ivy.linear`, the execution moves to torch backend, so I thought the decorator is to be added there also. Initially my approach was same as Ved mentioned, but it wasn't moving the arrays to gpu as expected, and it worked only when the decorator is present in the backend function as well. Let me know your thoughts on this.

Reply:
Adding `_auto_shift_device.auto_shift_device = True` to the wrapper solved this issue :smile: